ross/defects/crack.py

```diff
@@ -0,0 +1,603 @@
+
+    def _gasch(self):
+        # Gasch
+        kme = (self.ko + self.kcx) / 2
+        kmn = (self.ko + self.kcz) / 2
+        kde = (self.ko - self.kcx) / 2
+        kdn = (self.ko - self.kcz) / 2
+
+        kee = kme + (4 / np.pi) * kde * (
+            np.cos(self.angular_position)
+            - np.cos(3 * self.angular_position) / 3
+            + np.cos(5 * self.angular_position) / 5
+            - np.cos(7 * self.angular_position) / 7
+            + np.cos(9 * self.angular_position) / 9
+            - np.cos(11 * self.angular_position) / 11
+            + np.cos(13 * self.angular_position) / 13
+            - np.cos(15 * self.angular_position) / 15
+            + np.cos(17 * self.angular_position) / 17
+            - np.cos(19 * self.angular_position) / 19
+            + np.cos(21 * self.angular_position) / 21
+            - np.cos(23 * self.angular_position) / 23
+            + np.cos(25 * self.angular_position) / 25
+            - np.cos(27 * self.angular_position) / 27
+            + np.cos(29 * self.angular_position) / 29
+            - np.cos(31 * self.angular_position) / 31
+            + np.cos(33 * self.angular_position) / 33
+            - np.cos(35 * self.angular_position) / 35
+        )
+
+        knn = kmn + (4 / np.pi) * kdn * (
+            np.cos(self.angular_position)
+            - np.cos(3 * self.angular_position) / 3
+            + np.cos(5 * self.angular_position) / 5
+            - np.cos(7 * self.angular_position) / 7
+            + np.cos(9 * self.angular_position) / 9
+            - np.cos(11 * self.angular_position) / 11
+            + np.cos(13 * self.angular_position) / 13
+            - np.cos(15 * self.angular_position) / 15
+            + np.cos(17 * self.angular_position) / 17
+            - np.cos(19 * self.angular_position) / 19
+            + np.cos(21 * self.angular_position) / 21
+            - np.cos(23 * self.angular_position) / 23
+            + np.cos(25 * self.angular_position) / 25
+            - np.cos(27 * self.angular_position) / 27
+            + np.cos(29 * self.angular_position) / 29
+            - np.cos(31 * self.angular_position) / 31
+            + np.cos(33 * self.angular_position) / 33
+            - np.cos(35 * self.angular_position) / 35
+        )
+
+        aux = np.array([[kee, 0], [0, knn]])
+
+        K = ((self.T_matrix.T).dot(aux)).dot(self.T_matrix)
+
+        return K
+
+    def _mayes(self):
+        # Mayes
+
+        kee = 0.5 * (self.ko + self.kcx) + 0.5 * (self.ko - self.kcx) * np.cos(
+            self.angular_position
+        )
+
+        knn = 0.5 * (self.ko + self.kcz) + 0.5 * (self.ko - self.kcz) * np.cos(
+            self.angular_position
+        )
+
+        aux = np.array([[kee, 0], [0, knn]])
+
+        K = ((self.T_matrix.T).dot(aux)).dot(self.T_matrix)
+
+        return K
+
+    def _get_coefs(self, coef):
+        x = scipy.io.loadmat(
+            "/home/izabela/Documents/Projeto EDGE Petro/ross/PAPADOPOULOS_c"
+        )
```

Comment:
I recommend using the pathlib library here (considering that the .mat file is in the same directory as the crack.py):
```python
from pathlib import Path

file = Path(__file__).parent / "PAPADOPOULOS_c.mat"
x = scipy.io.loadmat(file)
```

Reply:
Agreed, this is still a part under development. We moved our efforts to finish the misalignment problem first, and left the cracks and the rubbing defects in standby, so a number of small problems like this should still be there.